src/makernote_int.cpp

```diff
@@ -1183,18 +1183,12 @@ namespace Exiv2 {
     }
     int sony2010eSelector(uint16_t /*tag*/, const byte* /*pData*/, uint32_t /*size*/, TiffComponent* const pRoot)
     {
-        const char* models[] = { "SLT-A58", "SLT-A99", "ILCE-3000", "ILCE-3500", "NEX-3N", "NEX-5R", "NEX-5T", "NEX-6", "VG30E", "VG900",
-            "DSC-RX100", "DSC-RX1", "DSC-RX1R", "DSC-HX300", "DSC-HX50V", "DSC-TX30", "DSC-WX60", "DSC-WX200", "DSC-WX300" };
-        std::set<std::string> s2010eModels;
-        for (auto&& model : models) {
-            s2010eModels.insert(model);
-        }
-        std::string model = getExifModel(pRoot);
-        int idx = -1;
-        if (s2010eModels.find(model) != s2010eModels.end()) {
-            idx = 0;
-        }
-        return idx;
+        static constexpr const char* models[] = {
```

Comment:
What do you think about turning this into a `std::array` instead? 

An additional small suggestion: I think that using `std::any_of` is a bit more aligned to what the actual intent here is. 
e.g.
```cpp
static constexpr std::array<std::string,19> models = {...};
auto const model = getExifModel(pRoot); 
return std::any_of(begin(models), end(models), [&model](auto const& elem){return elem==model;} ? 0 : -1;
```

Reply:
std::array requires a size parameter (except with C++17 where you have CTAD). Insertion and deletion of members can have negative consequences if the size is not updated. Best to keep it automatic.

std::any_of is not ideal as it requires a lambda. auto is Kansas is a C++14 feature. std::find does the same thing anyway.

Reply:
Re std array, I'd love for us to move to c++17, but that's a discussion for a different issue :D  
Even though you have to now specify the size, as part of the modernization effort we should surely try and get rid of c-style constructs. 

Code above was only to showcase what I meant, you don't have to use `auto` in the lambda... 
I also don't think using a lambda is problematic.  

The intent of the check is: "is model in models", which is what `any_of` does and is thus IMHO more self documenting than the `find` version which returns an extra iterator that you then still have to compare to the end to figure out if you found something...

Reply:
Getting rid of C constructs would be lovely. But only where it makes sense. std::array is too error prone before C++17. std:initializer_list at first glance looks like a solution, but it's actually a view into an array, not an array itself. I've had lots of fun taking addresses of tempories and wondering why nothing works.

I disagree. std::any_of internally is the same as std::find_if != std::end. There's no need to create a lambda where passing the value itself is sufficient.

Reply:
It's not about what it is internally, I don't think this code is performance critical enough to start that discussion.     
It's about what you convey to the future reader that has to understand code.  

And reading `any_of` tells you exactly what this function is doing.

Reply:
I like to avoid lambdas if possible. It's extra complexity for little benefit. std::find != std::end is also shorter.

As an example: https://github.com/gerbera/gerbera/commit/74dffe64bcfe265c50530d55ac4d0dc94e0d90f9

Reply:
I remain unconvinced... but in any case, it seems like we just have different points of view here, fair enough.   
If @piponazo thinks the current version is ok, I'm happy to move on.